src/clr-angular/data/datagrid/datagrid-action-overflow.ts

```diff
@@ -50,6 +76,21 @@ export class ClrDatagridActionOverflow implements OnDestroy {
     if (boolOpen !== this._open) {
       this._open = boolOpen;
       this.openChanged.emit(boolOpen);
+      if (boolOpen && isPlatformBrowser(this.platformId)) {
+        this.zone.runOutsideAngular(() => {
+          // It emits 2+ onStable events in a synchronous sequence. We need to wait for the last one.
+          // Some of them are probably only in dev mode, but some are caused by collapsing the previous overflow window.
+          this.zone.onStable
```

Comment:
This might seem like :trollface: request. But can't we just use setTimeout here? Because it will run outside of angular zone so it won't trigger any change detection, and no other method is dependent on that call, but only Element.focus().

Reply:
:) I don't consider it :trollface: 
I was also first thinking about `setTimeout`, but `zone.onStable` looks like a bit more specific and controlled lifecycle handler that still does the job. While `setTimeout` is a heavyweight weapon. I don't mind using it, but I would like to hear more opinions before I switch to it.
`Why` would we prefer `setTimeout` over `zone.onStable`?

Reply:
Ultimately, we don't want to be linked to Zones because long term they might get removed from Angular (Ivy makes this potentially a thing), and as you can see the API for this is...cumbersome.

Reply:
The main drawback I saw here was it currently fires twice and we are picking 2nd one to subscribe. After introducing changes somewhere, what if it starts emits more than that, and the 2nd becomes not what we want to subscribe to any longer? On the contrast, `setTimeout` is called outside of the zone, so we don't need to worry about triggering change detection. Secondly, we know it will run only once and after all microtasks so there's more assurance there. After all, it's also a pretty common pattern to call setTimeout outside of the zone. Plus, I think (it might be subjective 😛 ) it's is concise and non-complex readability-wise.

Reply:
Just for the record, we're taking the last of the synchronous series (that's what the debounceTime(0) stands for), second, third, fifth, no matter.
I agree to the statement that we want to keep away from zone intricacies when their future is uncertain. So I am leaving the "runOutsideAngular" but removing the confusing "onStable".
Thanks for the comments!